File: assignments/setup-jax-rs/henry/src/main/java/org/henrypost/itm466/model/Message.java
package org.henrypost.itm466.model;

import javax.xml.bind.annotation.XmlRootElement;
import java.util.Date;

@XmlRootElement
public class Message {

    private long id;
    private String message;
    private String author;
    private Date created;

    public Message() {

    }

    public Message(long _id, String _message, String _author) {
        id = _id;
        message = _message;
        author = _author;
        created = new Date();
    }

    public Message(long _id, String _message, String _author, Date _created) {
        id = _id;
        message = _message;
        author = _author;
        created = _created;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }
}

File: assignments/setup-jax-rs/henry/src/main/java/org/henrypost/itm466/model/Message.kt
package org.henrypost.itm466.model

import java.util.*
import javax.xml.bind.annotation.XmlRootElement
import javax.xml.bind.annotation.XmlTransient
import kotlin.collections.ArrayList
import kotlin.collections.HashMap

@XmlRootElement
class Message {

    var id: Long? = null
    var message: String? = null
    var author: String? = null
    var created: Date? = null
    private var comments: Map<Long, Comment> = HashMap()
    var links: List<Link> = ArrayList()

    constructor() {

    }


    constructor(_id: Long, _message: String, _author: String) {
        this.id = _id
        this.message = _message
        this.author = _author
        created = Date()
    }

    constructor(_id: Long, _message: String, _author: String, _created: Date) {
        this.id = _id
        this.message = _message
        this.author = _author
        created = _created
    }

    @XmlTransient
    public fun getComments(): Map<Long, Comment> {
        return comments
    }

    public fun setComments(comments: Map<Long, Comment>) {
        this.comments = comments
    }

    public fun addLink(url: String, rel: String) {
        val link = Link()
        link.link = url
        link.rel = rel
        links += link
    }

}
